src/python/WMCore/WMSpec/StdSpecs/Resubmission.py

```diff
@@ -39,15 +39,18 @@ def buildWorkload(self, arguments):
         helper.ignoreOutputModules(self.ignoredOutputModules)
 
         # override a couple of parameters, if provided by user
-        if 'Memory' in arguments:
-            helper.setMemory(arguments['Memory'])
-        if 'Campaign' in arguments and not helper.getCampaign():
-            helper.setCampaign(arguments["Campaign"])
         if 'RequestPriority' in arguments:
             helper.setPriority(arguments["RequestPriority"])
-        if 'TimePerEvent' in arguments:
-            for task in helper.taskIterator():
-                task.setJobResourceInformation(timePerEvent=arguments["TimePerEvent"])
+        if arguments['OriginalRequestName'] == 'TaskChain' and not isinstance(arguments['Memory'], dict):
+            # Memory must be a dict object if user wants to override TaskChain tasks
+            pass
+        else:
+            helper.setMemory(arguments['Memory'])
```

Comment:
How about changing to following? It is more compact but if you think it might be less clear, don't change.
```
if arguments['OriginalRequestName'] != 'TaskChain' or isinstance(arguments['Memory'], dict):
    helper.setMemory(arguments['Memory'])
```